### PR TITLE
break(amplify_auth_cognito): fixes getCurrentUser disparity

### PR DESCRIPTION
**Pull request by haverchuck**
*Description of changes:*
This PR should not be merged until [this android sdk PR ](https://github.com/aws-amplify/aws-sdk-android/pull/2555)is merged.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by Jordan-Nelson — APPROVED** [review]
The changes LGTM

File: packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthCognito.kt [hunk 1]
```diff
@@ -383,13 +384,37 @@ public class AuthCognito : FlutterPlugin, ActivityAware, MethodCallHandler, Plug
...
+        we need to use the mobileclient to get the old username to enforce platform parity.
+      */
+      Amplify.Auth.fetchAuthSession(
+        { result ->
+          val cognitoAuthSession = result as AWSCognitoAuthSession
+          when (cognitoAuthSession.userSub.type) {
+            // If the user sub accessor successful the user is present and the session should be valid
+            AuthSessionResult.Type.SUCCESS -> {
+              var awsMobileClient = AWSMobileClient.getInstance()
+              var username = awsMobileClient.username
```
Comment (HuiSF) [inline packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthCognito.kt:398]: nit: use `val` if the variable won't be reassigned

File: packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthCognito.kt [hunk 2]
```diff
@@ -383,13 +384,37 @@ public class AuthCognito : FlutterPlugin, ActivityAware, MethodCallHandler, Plug
...
+              prepareUserResult(flutterResult, AuthUser(cognitoAuthSession.userSub.toString(), username));
+            }
+            // If the user sub accessor failed, check the signIn state
+            AuthSessionResult.Type.FAILURE -> {
+              // if signedIn flag still true, we know session is expired so get the old user
+              if (result.isSignedIn) {
+                var awsMobileClient = AWSMobileClient.getInstance()
+                var username = awsMobileClient.username
+                var userid = awsMobileClient.userSub
```
Comment (HuiSF) [inline packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthCognito.kt:407]: nit: ditto `val` vs `var`

**Review by HuiSF — APPROVED** [review]
Looks good, left comments re: what could be warned while compiling.

**Pull request merged**